pkg/metrics/sampler/matcher.go

```diff
@@ -277,70 +262,76 @@ func (ne constantMatcher) String() string {
 	return fmt.Sprint(ne.value)
 }
 
-// NewSampleMatchFn creates new includeSampleMatchFn func, enableProcessMetrics might be nil when
-// value was not set.
-func NewSampleMatchFn(enableProcessMetrics *bool, metricsMatchers config.MetricsMap, ffRetriever feature_flags.Retriever) MatcherFn {
-	// configuration option always takes precedence over FF and matchers configuration
-	if enableProcessMetrics == nil {
-		matcher := matcherFromMetricsMatchers(metricsMatchers)
-		if matcher != nil {
-			return matcher
+// NewIncludeSampleMatchFn returns a function `func(sample) bool` that determinesif the sample
+// should be included (true) as an event or not (false). Note that this is NOT the negation
+// of `NewExcludeSampleMatchFn`.
+func NewIncludeSampleMatchFn(enableProcessMetrics *bool, metricsMatchers config.IncludeMetricsMap, ffRetriever feature_flags.Retriever) MatcherFn {
+	return func(sample any) bool {
+		// We return early if the sample is not a ProcessSample.
+		if !isProcessSample(sample) {
+			// Include this event
+			return true
 		}
```

Comment:
I think this function is not well designed from its inception since it's applied to all Samples but inside all the logic is for ProcessSample, so there should have been some way to only appy the function to that sample, but better to not modify much and do the change you did checking it its not a processSample and doing an early exit, good job.